lib/sycamore/sycamore/tests/unit/writers/test_base.py

```diff
@@ -0,0 +1,101 @@
+from dataclasses import dataclass
+from pathlib import Path
+from sycamore.data.document import Document, MetadataDocument
+from sycamore.plan_nodes import Node
+from sycamore.writers.base import BaseDBWriter
+
+
+class FakeClient(BaseDBWriter.Client):
+    def __init__(self, client_params: "FakeClientParams"):
+        self.fspath = client_params.fspath
+
+    @classmethod
+    def from_client_params(cls, params: "BaseDBWriter.ClientParams") -> "FakeClient":
+        assert isinstance(params, FakeClientParams)
+        return FakeClient(params)
+
+    def write_many_records(self, records: list["BaseDBWriter.Record"], target_params: "BaseDBWriter.TargetParams"):
+        for r in records:
+            assert isinstance(r, FakeRecord) and isinstance(target_params, FakeTargetParams)
+            file = self.fspath / target_params.dirname / r.doc_id
+            file.write_text(r.text)
+
+    def create_target_idempotent(self, target_params: "BaseDBWriter.TargetParams"):
```

Comment:
Suggest adding test for existing not matching params by adding mode to the params.